streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -763,11 +763,19 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
             return thread;
         });
 
-        rocksDBMetricsRecordingTriggerThread = Executors.newSingleThreadScheduledExecutor(r -> {
-            final Thread thread = new Thread(r, clientId + "-RocksDBMetricsRecordingTrigger");
-            thread.setDaemon(true);
-            return thread;
-        });
+        rocksDBMetricsRecordingTriggerThread = maybeGetRocksDBMetricsRecordingTriggerThread(clientId, config);
```

Comment:
super nit: maybe we should consider a different name since this not a `Thread` but `ScheduledExecutorService`? Something like `rocksDBMetricsExecutorService` ?   It's a subjective comment, so I'll leave up to you if you want to address this or not.

Reply:
Will rename it to `rocksDBMetricsRecordingService`.